### PR TITLE
feat(ci): included dependency review action

### PR DESCRIPTION
**Pull request by naveensrinivasan**
## Description

- Included dependency review action https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-dependency-review#dependency-review-enforcement

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: .github/workflows/dependency-review.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: 'Dependency Review'
+on: [pull_request]
+
+permissions:
+  contents: read
+
+jobs:
+  dependency-review:
```
Comment (Racer159) [inline .github/workflows/dependency-review.yml:8]:
```suggestion
  validate:
```
Changing the job name to validate will make it match our branch protection rules for the repo - we should also consider if/how we might de-conflict this dep review with `.github/workflows/scan-cves.yml` which does similar things (not license checks) but more manually.  We could leave both running to see which output we like better though over time (`grype` has gotten better but had a lot of false positives when we first started).

**Review by Racer159 — CHANGES_REQUESTED** [review]

**Review by Racer159 — APPROVED** [review]
lgtm

**Comment by Racer159** [comment]
Code looks good but some commits are not signed - will need to be resigned to merge
image

**Review by Racer159 — CHANGES_REQUESTED** [review]
There is still this unsigned commit in there: 
image

**Comment by naveensrinivasan** [comment]
> There is still this unsigned commit in there: image

Can you please check now?

**Comment by Noxsios** [comment]
Approving. Will need to circle back in a few weeks/months to evaluate how much of a value add / noise this generates.

File: .github/workflows/dependency-review.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+name: 'Dependency Review'
+on: [pull_request]
```
Comment (Noxsios) [inline .github/workflows/dependency-review.yml:2]:
```suggestion
name: Dependency Review
on: pull_request
```

tiny consistency tweak

File: .github/workflows/dependency-review.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+permissions:
+  contents: read
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Checkout Repository'
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
+      - name: 'Dependency Review'
```
Comment (Noxsios) [inline .github/workflows/dependency-review.yml:13]:
```suggestion
      - name: Checkout
        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
      - name: Dependency Review
```

**Review by Noxsios — CHANGES_REQUESTED** [review]

**Review by Racer159 — APPROVED** [review]
lgtm

**Review by Noxsios — APPROVED** [review]

**Pull request merged**